### PR TITLE
fix release web builds where the target file is not under lib

### PR DESCRIPTION
**Pull request by jonahwilliams**
## Description

If the target file is not under lib, we will currently fail to build an app as we can't find a package uri. Fall back to a file uri in this case, which is okay since that library will probably have a package import.

### CODE REVIEW

File: packages/flutter_tools/lib/src/build_system/targets/web.dart [hunk 1]
```diff
@@ -68,11 +68,10 @@ class WebEntrypointTarget extends Target {
     );
 
     // By construction, this will only be null if the .packages file does not
-    // have an entry for the user's application.
-    final Uri mainImport = packageUriMapper.map(importPath);
-    if (mainImport == null) {
-      throw Exception('Missing package definition for $mainImport');
-    }
+    // have an entry for the user's application or if the main file is
+    // outisde of the lib/ directory.
```
Comment (yjbanov) [inline packages/flutter_tools/lib/src/build_system/targets/web.dart]: typo: outisde

File: packages/flutter_tools/lib/src/build_system/targets/web.dart [hunk 2]
```diff
@@ -68,11 +68,10 @@ class WebEntrypointTarget extends Target {
...
+    final String mainImport = packageUriMapper.map(importPath)?.toString()
+      ?? fs.file(importPath).absolute.uri.toString();
```
Comment (yjbanov) [inline packages/flutter_tools/lib/src/build_system/targets/web.dart]: Does this work on Windows?

**Review by yjbanov — APPROVED** [review]

File: packages/flutter_tools/lib/src/build_system/targets/web.dart (hunk 2 above)
Comment (jonahwilliams) [reply packages/flutter_tools/lib/src/build_system/targets/web.dart]: Yup!

File: packages/flutter_tools/lib/src/build_system/targets/web.dart (hunk 1 above)
Comment (jonahwilliams) [reply packages/flutter_tools/lib/src/build_system/targets/web.dart]: Fixed

**Pull request merged**